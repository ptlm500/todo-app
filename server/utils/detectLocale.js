const acceptLanguage = require('accept-language');
const config = require('../config.json');

function detectLocale(req) {
  acceptLanguage.languages(config.ACCEPTED_LANGUAGES);

  const cookieLocale = req.cookies.locale;

  return acceptLanguage.get(cookieLocale || req.headers['accept-language']) ||
    config.DEFAULT_LOCALE;
}

module.exports = detectLocale;
