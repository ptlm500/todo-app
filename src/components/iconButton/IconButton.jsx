import styled from 'styled-components';
import PropTypes from 'prop-types';
import { theme } from '../../styles';

const LARGE = 'large';
const SMALL = 'small';
const LARGE_SIZE = '3rem';
const SMALL_SIZE = '1rem';
const sizes = {
  'large': LARGE_SIZE,
  'small': SMALL_SIZE
};

function getSize(size) {
  return sizes[size];
}

const IconButton = styled.button`
  background: none;
  border: none;
  width: ${props => getSize(props.size)};
  height: ${props => getSize(props.size)};
  padding: 0;
  cursor: pointer;
  transition: ${theme.transition.long};

  > svg {
    width: 100%;
    height: 100%;
  }

  &:hover {
    transform: scale(1.2) rotate(1turn);
  }
`;

IconButton.propTypes = {
  size: PropTypes.oneOf([LARGE, SMALL])
};

IconButton.defaultProps = {
  size: LARGE
};

export default IconButton;
