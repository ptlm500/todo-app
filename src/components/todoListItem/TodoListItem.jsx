import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { removeTodo } from '../../state/actions/actions';

export function TodoListItem(props) {
  const {
    id,
    name,
    description,
    creationDate,
    onRemove
  } = props;

  function remove() {
    onRemove(id);
  }

  return (
    <li className='todo-list__item'>
      <div>
        <h3>{name}</h3>
        <button onClick={remove}>
          Remove
        </button>
      </div>
      <a>{description}</a>
      <a>{creationDate}</a>
    </li>
  );
}

TodoListItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  creationDate: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => {
  return {
    onRemove: id => dispatch(removeTodo(id))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(TodoListItem);
