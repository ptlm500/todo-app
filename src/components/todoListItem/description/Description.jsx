import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import InlineEditable from '../../inline-editable';
import AutosizingTextarea from '../../autosizingTextarea';
import { theme } from '../../../styles';

function Description(props) {
  const {
    className,
    updateDescription
  } = props;
  const descriptionInStore = props.description;

  const [description, setDescription] = useState(descriptionInStore);

  useEffect(() => {
    // Set description state if stored description changes
    setDescription(descriptionInStore);
  }, [descriptionInStore]);

  return (
    <InlineEditable
      className={className}
      value={description}
      placeholder={' '}
      onSubmit={updateDescription}
    >
      <AutosizingTextarea
        onChange={e => setDescription(e.target.value)}
      />
    </InlineEditable>
  );
}

Description.propTypes = {
  className: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  updateDescription: PropTypes.func.isRequired
};

export default styled(Description)`
  ${theme.bodyFontSize};
  margin-bottom: 1rem;
  min-width: 100px;
`;
