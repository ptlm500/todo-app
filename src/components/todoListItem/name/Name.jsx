import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import InlineEditable from '../../inline-editable';
import Input from '../../input';
import { theme } from '../../../styles';


function Name(props) {
  const {
    className,
    updateName
  } = props;
  const nameInStore = props.name;
  const [name, setName] = useState(nameInStore);

  useEffect(() => {
    // Set name state if stored name changes
    setName(nameInStore);
  }, [nameInStore]);

  return (
    <InlineEditable
      className={className}
      value={name}
      placeholder={' '}
      onSubmit={updateName}
    >
      <Input
        onChange={e => setName(e.target.value)}
      />
    </InlineEditable>
  );
}

Name.propTypes = {
  className: PropTypes.string.isRequired,
  updateName: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
};

export default styled(Name)`
  ${theme.headerFontSize};
  font-weight: 600;
  min-width: 100px;
`;
