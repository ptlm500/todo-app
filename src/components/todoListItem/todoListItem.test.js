import React from 'react';
import { act } from 'react-test-renderer';
import { renderWithIntl } from '../../../test/utils';
import { TodoListItem } from './TodoListItem';
import IconButton from '../iconButton';
import Name from './name';
import Description from './Description';

const defaultProps = {
  id: 'test-id',
  name: 'Test name',
  description: 'test description',
  creationDate: '1582043913000',
  onRemove: jest.fn(),
  onUpdate: jest.fn(),
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('TodoListItem', () => {
  it('renders a todoListItem matching the snapshot', () => {
    let tree;
    act(() => {
      tree = renderWithIntl(<TodoListItem {...defaultProps}/>);
    });

    expect(tree).toMatchSnapshot();
  });

  it('calls onRemove when the remove button is clicked', () => {
    let tree;
    act(() => {
      tree = renderWithIntl(<TodoListItem {...defaultProps}/>);
    });

    const removeButton = tree.root.findByType(IconButton);
    removeButton.props.onClick();

    expect(defaultProps.onRemove).toHaveBeenCalledWith(defaultProps.id);
  });

  it('calls onUpdate correctly for the Name component', () => {
    let tree;
    act(() => {
      tree = renderWithIntl(<TodoListItem {...defaultProps}/>);
    });

    const name = tree.root.findByType(Name);

    act(() => {
      name.props.updateName(defaultProps.name);
    });
    // name is equal, so we shouldn't be trying to update
    expect(defaultProps.onUpdate).toHaveBeenCalledTimes(0);

    const newName = 'New name';
    act(() => {
      name.props.updateName(newName);
    });
    // new name, so we should update
    expect(defaultProps.onUpdate).toHaveBeenCalledWith({
      id: defaultProps.id,
      name: newName
    });
  });

  it('calls onUpdate correctly for the Description component', () => {
    let tree;
    act(() => {
      tree = renderWithIntl(<TodoListItem {...defaultProps}/>);
    });

    const description = tree.root.findByType(Description);

    act(() => {
      description.props.updateDescription(defaultProps.description);
    });
    // description is equal, so we shouldn't be trying to update
    expect(defaultProps.onUpdate).toHaveBeenCalledTimes(0);

    const newDescription = 'New description';
    act(() => {
      description.props.updateDescription(newDescription);
    });
    // new description, so we should update
    expect(defaultProps.onUpdate).toHaveBeenCalledWith({
      id: defaultProps.id,
      description: newDescription
    });
  });
});
