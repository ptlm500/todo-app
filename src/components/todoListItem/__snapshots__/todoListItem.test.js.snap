// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem renders a todoListItem matching the snapshot 1`] = `
.c6 {
  font-size: 2.5vw;
  color: #757575;
}

.c3 {
  white-space: pre-wrap;
  word-break: break-word;
}

.c5 {
  font-size: 2.5vw;
  margin-bottom: 1rem;
  min-width: 100px;
}

.c4 {
  background: none;
  border: none;
  width: 1rem;
  height: 1rem;
  padding: 0;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c4 > svg {
  width: 100%;
  height: 100%;
}

.c4:hover {
  -webkit-transform: scale(1.2) rotate(1turn);
  -ms-transform: scale(1.2) rotate(1turn);
  transform: scale(1.2) rotate(1turn);
}

.c2 {
  font-size: 5vw;
  font-weight: 600;
  min-width: 100px;
}

.c0 {
  background: #C5CAE9;
  border-radius: 0.5rem;
  padding: 1rem;
  border: 1px solid #3F51B5;
  min-width: 450px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c0.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1rem;
}

@media (max-width:480px) {
  .c6 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c6 {
    font-size: 16px;
  }
}

@media (max-width:480px) {
  .c5 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c5 {
    font-size: 16px;
  }
}

@media (max-width:240px) {
  .c2 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c2 {
    font-size: 32px;
  }
}

@media (max-width:599px) {
  .c0 {
    min-width: 300px;
    max-width: 90%;
  }
}

<div
  className="sc-AxjAm sc-AxirZ c0"
>
  <div
    className="c1"
  >
    <div
      className="c2 inline-editable"
    >
      <div
        className="inline-editable__edit-trigger"
        onClick={[Function]}
      >
        <span
          className="c3"
        >
          Test name
        </span>
      </div>
    </div>
    <button
      className="c4"
      onClick={[Function]}
      size="small"
    >
      <svg
        height="18px"
        version="1.1"
        viewBox="0 0 14 18"
        width="14px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Delete
        </title>
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <g
            transform="translate(-207.000000, -245.000000)"
          >
            <g
              transform="translate(100.000000, 100.000000)"
            >
              <g
                id="Outlined-/-Action-/-delete"
                transform="translate(102.000000, 142.000000)"
              >
                <g>
                  <polygon
                    id="Path"
                    points="0 0 24 0 24 24 0 24"
                  />
                  <path
                    d="M16,9 L16,19 L8,19 L8,9 L16,9 Z M14.5,3 L9.5,3 L8.5,4 L5,4 L5,6 L19,6 L19,4 L15.5,4 L14.5,3 Z M18,7 L6,7 L6,19 C6,20.1 6.9,21 8,21 L16,21 C17.1,21 18,20.1 18,19 L18,7 Z"
                    fill="currentColor"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    </button>
  </div>
  <div
    className="c5 inline-editable"
  >
    <div
      className="inline-editable__edit-trigger"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        test description
      </span>
    </div>
  </div>
  <a
    className="c6"
  >
    February 18, 2020, 4:38 PM
  </a>
</div>
`;
